hbase-server/src/main/java/org/apache/hadoop/hbase/master/HMaster.java

```diff
@@ -2275,6 +2275,29 @@ protected String getDescription() {
       });
   }
 
+  @Override
+  public long truncateRegion(final RegionInfo regionInfo, final long nonceGroup, final long nonce)
```

Comment:
1.Added the pre/post hooks.
2.truncateRegion and truncateTable looks similar because they following the pattern preHook->submitProcedure->postHook. But internally the workflow is different. There are other methods like deleteTable also similar. We could add

Reply:
Yeah, the whole block under "submitProcedure" is identical, apart from the procedure we are creating and specific pre/post hooks. But am ok to leave this work for a separate PR, as there are few others in the same style.